app/src/main/java/io/neurolab/activities/MeditationHome.java

```diff
@@ -30,6 +31,21 @@ protected void onCreate(Bundle savedInstanceState) {
         setMeditationCategoryIntent(travelView);
         setMeditationCategoryIntent(shBreakView);
         setMeditationCategoryIntent(lgBreakView);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    public boolean onSupportNavigateUp(){
+        onBackPressed();
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        startActivity(new Intent(this, NeuroLab.class));
+        finish();
```

Comment:
Avoid closing the activities rather than firing intents when pressing back

Reply:
@adityastic  Ok! I will change accordingly.

Reply:
Update this as well